src/gui/qgsgui.h

```diff
@@ -298,9 +298,22 @@ class GUI_EXPORT QgsGui : public QObject
      * The lambda will be run either if macros are currently allowed or if the user accepts the message.
      * The \a messageBar must be given in such case.
      * \param messageBar the message bar must be provided if a lambda method is used.
+     *
+     * \note Not available in Python bindings
      */
     static bool pythonMacroAllowed( void ( *lambda )() = nullptr, QgsMessageBar *messageBar = nullptr ) SIP_SKIP;
 
+    /**
+     * Returns TRUE if python expression functions from project are currently allowed to be loaded.
+     *
+     * If the global option is to ask user, a button to enable them will be shown in a message bar.
+     * \param messageBar Message bar to communicate with the user.
+     *
+     * \note Not available in Python bindings
+     * \since QGIS 3.40
+     */
+    static bool pythonExpressionFromProjectAllowed( QgsMessageBar *messageBar ) SIP_SKIP;
```

Comment:
Since we share the same setting used by macros, could we use a single function instead? We could remove/rename pythonMacroAllowed (since it's not part of the bindings, we can break the API here) and come up with a more generic name instead of adding a function that's only referring to another embedded code subset.

E.g.:

```suggestion
    static bool pythonEmbeddedInProjectAllowed( QgsMessageBar *messageBar ) SIP_SKIP;
```

You could then have an enum parameter in there to specify which subset is triggering the permission request and tweak the message box strings. But that'd at least reduce the code logic duplication.

Reply:
Nice suggestion, thank you!
We're now using the same function with an Enum parameter.
Implemented in https://github.com/qgis/QGIS/pull/58354/commits/c5286e4dcc86dd78f445e31dd70006240c1ef935